### PR TITLE
Fix CoffeeScript CJS + ESM

### PR DESCRIPTION
**Pull request by STRd6**
Added call to `module._compile` to fix CJS loading in combo mode.

Also refactored out `compile` helper method for compiling CoffeeScript.

### CODE REVIEW

File: coffeescript-loader/test.js [hunk 1]
```diff
@@ -24,12 +22,7 @@ child.stderr.on('data', (data) => {
...
-  ok(stdout.includes('Hello from CoffeeScript', 'Main entry transpiles'));
+  ok(stdout.includes('Hello from CoffeeScript'), 'Main entry transpiles');
   ok(stdout.includes('HELLO FROM ESM'), 'ESM import transpiles');
-
-  // There is a known issue between ESM + CJS where CJS named exports get lost:
-  // require.extentions appropriately supplies (transformed) source, but an
-  // empty object is returned for the module's exports.
-  // the import does NOT hit ESMLoader or its CJS strategy.
-  ok(!stdout.includes('Hello from CommonJS!'), 'Named CommonJS import fails');
+  ok(stdout.includes('Hello from CommonJS!'), 'Named CommonJS import fails');
```
Comment (GeoffreyBooth) [inline coffeescript-loader/test.js:27]: Does this pass now?

Comment (STRd6) [reply coffeescript-loader/test.js:27]: It does

Comment (GeoffreyBooth) [reply coffeescript-loader/test.js:27]: And what did the trick was adding `module._compile`, it looks like?

File: coffeescript-loader/loader.js [hunk 2]
```diff
@@ -5,7 +5,10 @@ import { dirname, extname, resolve as resolvePath } from 'path';
 import { fileURLToPath, pathToFileURL } from 'url';
 
 import CoffeeScript from 'coffeescript';
-
+const compile = (source, filename) => CoffeeScript.compile(source, {
```
Comment (GeoffreyBooth) [inline coffeescript-loader/loader.js]: Can we call this `coffeeCompile` or something to distinguish it from `module._compile`?

File: coffeescript-loader/test.js (hunk 1 above)
Comment (STRd6) [reply coffeescript-loader/test.js:27]: Yes adding `module._compile` was the critical piece. `coffeescript/register` called it in their implementation so I looked to see what was different here.

**Pull request merged**